hw07.c

```diff
@@ -0,0 +1,246 @@
+int tga_width( const TGAImage * self ) {
+    return tgaheader_width( &self->header );
+}
+
+int tga_height( const TGAImage * self ) {
+    return tgaheader_height( &self->header );
+}
+
+int tga_channels( const TGAImage * self ) {
+    return  tgaheader_channels( &self->header );
+}
+
+
+//prevodova fce z 2d souradnic do indexu 1D odpovidajiciho serioveho elementu
+int to1D(int row, int col, int cols){
+    return col + row * cols;
+}
+
+
+//cokoliv zavola fci se souradnicema x,y bude prevedeno z 2d pole do odpovidajiciho indexu 1d pole jiz nainacializovaneho v *data
+//prehodim x, y kdyz pocatek je 0,0 v dolnim levem rohy pokud prehodim x z y tak he to jakobych ho otozil o 90/ prohodil sloupce/radky
+void set_pixel(TGAImage *self, int y, int x, RGBA *pix){
+    self->data[to1D(x,y, tga_height(self))].alpha = pix->alpha;
+    self->data[to1D(x,y, tga_height(self))].blue = pix->blue;
+    self->data[to1D(x,y, tga_height(self))].green = pix->green;
+    self->data[to1D(x,y, tga_height(self))].red = pix->red;
+}
+
+void draw_bg(TGAImage *self, RGBA *bg){
+    for (int i = 0; i < tga_height(self); i++)
+    {
+        for (int j = 0; j < tga_width(self); j++)
+        {
+            set_pixel(self, i ,j , bg);
+            printf("bg: i:%d j:%d\n",i ,j);
+        }
+
+    }
+
+}
+
+//pocita indexy nalezejici vykresleni znaku  ty posila do fce ktere zapisuje ty hodnotu do pole clena strukty *data
+void draw_one(TGAImage *self, int dx, int dy, RGBA *pix){
+    for (int i = 0 + dx; i < 195 + dx; i++)
+    {
+        for (int j = 0 + dx; j < 169 + dy; j++)
+        {
+            if (i > 15 + dx && j > 139 + dy)
+            {
+                set_pixel(self, i, j, pix);
+                printf("one: i:%d j:%d\n",i ,j);
+            }
+        }
+    }
+}
+
+void draw_zero(TGAImage *self, int dx, int dy, RGBA *pix){
+    for (int i = 0 + dx; i < 195 + dx; i++)
+    {
+        for (int j = 0 + dx; j < 169 + dy; j++)
+        {
+            if (i > 15 + dx && j > 139 + dy)
+            {
+                set_pixel(self, i, j, pix);
+                printf("one: i:%d j:%d\n",i ,j);
+            }
+        }
+    }
+}
+
+//nevim jestli tohle muze byt jen v mainu nebo pokuud pouziju typedef nejak na tohle pak to jde takhle
+//typedef of array of function pointers (takhle pka func_array je pak data type plus jmesno)
+//typedef TGAImage* (*func_array[])(TGAImage *, size_t, size_t, RGBA *);
+
+void watch_draw_time(TGAImage* self, const int hours, const int minutes){
+
+    FILE *file = fopen("aw.tga","wb");
+    assert(file);
+
+    int firstH = hours / 10;;
+    int secondH = hours % 10;
+    int firstMin = minutes / 10;
+    int secondMin = minutes % 10;
+
+    RGBA fg = {.red = 30, .green = 10, .blue = 10, .alpha = 255};
+//zde musem volat fci s fcnim ukazatel na pole a pdle first/,... hodnot se zavola rovnou dana fce,,nepamatujeu se uz inicializaci atd
+//POLE FUNKCNICHU UKAZATELU
+    void (*func_array[])(TGAImage*, int, int, RGBA*) = {
```

Comment:
Tohle vypada jako totalni blbost. co se snazis udelat? Deklarovat funkci ve funkci je jakoze "big no-no". Nemas pro to libovolny duvod.

Reply:
Jakoze uz vidim, ale fakt predavat to jako funkci je dost ... blbost.

Reply:
jakoze to melo sedet na ta jednotliva digit a ty odpovidaji tem jednotlivim inicializacim toho pole ukazatelu na funkci a tak to rovnou zavola tu spravnou fci samo podle ruznejch cifer, aniz bych mel fory a casy a jediny o co se budu muset starat so ty offsety
ale slysim ze blbost teda